app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
+        # connection_by_role=(option)
+    end
+  end
+
+  def connections
```

Comment:
I agree on the naming of `connections` and by extension `connection_by_role`...even after reading this, I read the code and was thinking "oh, connection_by_role makes the connection", but no, it doesn't.

I'm thinking that if you are moving toward representing connection configuration in a struct, that maybe these methods should be name `connection_configurations`, `connection_configurations=` and `connection_configuration_by_role`